types.go

```diff
@@ -1765,6 +1766,7 @@ type Message struct {
 
 	// doneSignal is a channel that indicate when a message is considered acted upon by downstream handler
 	doneSignal chan struct{}
+	closeOnce  sync.Once
```

Comment:
This needs to be a `*sync.Once` to avoid copying the internal mutex (see CI failures).

Reply:
kept it as a value to avoid having to initialize it, and wrapped its usage to have it by reference instead.

Reply:
`go vet` is still complaining about this.